pkg/util/filesystem/util_windows.go

```diff
@@ -260,3 +261,79 @@ func Chmod(path string, filemode os.FileMode) error {
 func IsAbs(path string) bool {
 	return filepath.IsAbs(path) || strings.HasPrefix(path, `\`) || strings.HasPrefix(path, `/`)
 }
+
+// IsPathValidForMount checks if the given `path` is a valid mount point.
+// Return whether the path is valid and error encountered if any
+func IsPathValidForMount(path string) (bool, error) {
+	path = strings.TrimSuffix(path, string(filepath.Separator))
+	dir, err := os.Lstat(path)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return false, err
+		}
+
+		return false, nil
+	}
+
+	if dir.IsDir() {
+		// the path may not be mounted yet
+		return true, nil
+	}
+
+	// for windows NTFS, check if the path is symlink instead of directory.
+	isSymlink := dir.Mode()&os.ModeSymlink != 0
+
+	// mounted folder created by SetVolumeMountPoint may still report ModeSymlink == 0
+	mountedFolder, err := isMountedFolder(path)
+	if err != nil {
+		return false, err
+	}
+
+	if isSymlink || mountedFolder {
+		return true, nil
+	}
+
+	return false, fmt.Errorf("path %v exists but is not a mounted path", path)
+}
+
+// isMountedFolder checks whether the `path` is a mounted folder.
+func isMountedFolder(path string) (bool, error) {
```

Comment:
rename as `IsMountedFolder` since I am this function is useful in other places

Reply:
Sure.